src/Transport/Receiving/MessageReceiverNotifier.cs

```diff
@@ -154,6 +162,13 @@ public void Start()
             await AbandonInternal(message);
         }
 
+        async Task AbandonAsyncOnCancellation(BrokeredMessage message)
+        {
+            logger.Info("Received message is cancelled by the pipeline, abandoning it so we can process it later.");
+
+            await AbandonInternal(message);
```

Comment:
I can do this. That would be a bit different from already existing: AbandonAsync, InvokeCompletionCallbacksAsync though